### PR TITLE
Add role for managing hostgroups

### PR DESCRIPTION
**Pull request by ehelms**
Putting into draft initially as I want to work on a few better examples. Ideas welcome on what we can show here.

### CODE REVIEW

File: roles/hostgroups/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    puppet_proxy: puppet-proxy.example.com
+    puppet_ca_proxy: puppet-proxy.example.com
+    openscap_proxy: openscap-proxy.example.com
+  - name: "CentOS 7"
+    organization: "Default Organization"
+    lifecycle_environment: "Production"
+    content_view: "CentOS 7"
+    parameters:
+      - name: "kt_activation_keys"
+        value: "centos-7"
```
Comment (evgeni) [inline roles/hostgroups/README.md]: you can set `activation_keys: centos-7` now, which is easier to read :)

File: roles/hostgroups/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+---
+- name: 'Create Hostgroups'
+  theforeman.foreman.hostgroup:
+    username: "{{ username }}"
+    password: "{{ password }}"
+    server_url: "{{ server_url }}"
```
Comment (evgeni) [inline roles/hostgroups/tasks/main.yml]: `| default(omit)` :)

File: roles/hostgroups/README.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            puppet_proxy: puppet-proxy.example.com
+            puppet_ca_proxy: puppet-proxy.example.com
+            openscap_proxy: openscap-proxy.example.com
+          - name: "CentOS 7"
+            organization: "Default Organization"
+            lifecycle_environment: "Production"
+            content_view: "CentOS 7"
+            parameters:
+              - name: "kt_activation_keys"
+                value: "centos-7"
```
Comment (evgeni) [inline roles/hostgroups/README.md]: same as above

File: roles/hostgroups/README.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+- `hostgroups`: List of hostgroups to manage that are each represented as a dictionary. See module documentation for a list of available options for each hostgroup.
+  Hostgroups may have any set of fields defined on them and may optionally define a `parent` for nested hostgroups.
+  A variety of examples are demonstrated in the data structure below:
+
+```yaml
+hostgroups:
+  - name: "Basic example"
+    architecture: "x86_64"
+    operatingsystem: "CentOS"
+    medium: "partition_table_medium"
```
Comment (evgeni) [inline roles/hostgroups/README.md]: this sounds odd, did you mean "basic_medium" or something?

**Comment by evgeni** [comment]
few comments, but no good ideas about more examples

File: roles/hostgroups/README.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+```yaml
+- hosts: localhost
+  roles:
+    - role: theforeman.foreman.hostgroups
+      vars:
+        server_url: https://foreman.example.com
+        username: "admin"
+        password: "changeme"
+        organization: "Default Organization"
```
Comment (wbclark) [inline roles/hostgroups/README.md]:
```suggestion
        foreman_server_url: https://foreman.example.com
        foreman_username: "admin"
        foreman_password: "changeme"
        foreman_organization: "Default Organization"
```

Reminder that this convention should be used throughout :)

Comment (ehelms) [reply roles/hostgroups/README.md]: Good catch, updated

File: tests/test_playbooks/hostgroups_role.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  gather_facts: false
+  vars_files:
+    - vars/server.yml
+  roles:
+    - role: hostgroups
+      vars:
+        username: "{{ foreman_username }}"
+        password: "{{ foreman_password }}"
+        server_url: "{{  foreman_server_url }}"
+        validate_certs: "{{ foreman_validate_certs }}"
```
Comment (evgeni) [inline tests/test_playbooks/hostgroups_role.yml]:
```suggestion
```

these are unneeded, as they are already set in `server.yml` (and there with the correct name ;))

File: tests/test_playbooks/hostgroups_role.yml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        organization: "Test Organization"
```
Comment (evgeni) [inline tests/test_playbooks/hostgroups_role.yml]: this is also unused, it should be either *in* `foreman_hostgroups` or dropped.

**Review by evgeni — APPROVED** [review]
one tiny nit on the test, but nothing major/blocking

**Comment by evgeni** [comment]
aaah, changelog entry please XD

**Comment by ehelms** [comment]
Added

**Pull request merged**